src/libsyntax/parse/parser.rs

```diff
@@ -4107,7 +4107,7 @@ impl<'a> Parser<'a> {
         self.parse_ty_param_bounds_common(true)
     }
 
-    // Parse bounds of a type parameter `BOUND + BOUND + BOUND` without trailing `+`.
+    // Parse bounds of a type parameter `BOUND + BOUND + BOUND`.
```

Comment:
type parameter -> lifetime parameter

I'd also write "possibly with trailing `+`" to avoid future questions for sure.